### PR TITLE
IGNITE-8558 Provide an opportunity to stop grids and cancel tasks after execution all tests

### PR DESCRIPTION
**Pull request by Rodion4ik**
Add method to class GridAbstractTest
Add new boolean method "isWillJobsCancelled()" which can be overriden

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java [hunk 1]
```diff
@@ -1195,6 +1195,13 @@ protected void stopGrid(@Nullable String igniteInstanceName, boolean cancel, boo
             stopGridErr = true;
         }
     }
+    
+     /**
+     * Whether jobs will be cancelled forcibly before stopping grids
```
Comment (SomeFire) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java]:
```suggestion
     * Whether jobs will be cancelled forcibly on stopping grids.
```

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java [hunk 2]
```diff
@@ -1195,6 +1195,13 @@ protected void stopGrid(@Nullable String igniteInstanceName, boolean cancel, boo
...
+     */
+    protected boolean isWillJobsCancelled() {
```
Comment (SomeFire) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java]:
```suggestion
    protected boolean cancelJobsOnTearDown() {
```

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java [hunk 3]
```diff
@@ -1829,7 +1836,7 @@ protected static ClassLoader getExternalClassLoader() {
                 }
 
                 if (isSafeTopology()) {
-                    stopAllGrids(false);
+                    stopAllGrids(isWillJobsCancelled());
```
Comment (SomeFire) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java]:
```suggestion
                    stopAllGrids(cancelJobsOnTearDown());
```

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java [hunk 4]
```diff
@@ -1195,6 +1195,13 @@ protected void stopGrid(@Nullable String igniteInstanceName, boolean cancel, boo
             stopGridErr = true;
         }
     }
+    
+     /**
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java]:
```suggestion
    /**
```

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java [hunk 5]
```diff
@@ -1829,7 +1836,7 @@ protected static ClassLoader getExternalClassLoader() {
                 }
 
                 if (isSafeTopology()) {
-                    stopAllGrids(false);
+                    stopAllGrids(isCancelJobsAfterTestsStopped());
```
Comment (SomeFire) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java]:
```suggestion
                    stopAllGrids(cancelJobsAfterTestsStopped());
```

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxConcurrentRemoveObjectsTest.java [hunk 6]
```diff
@@ -72,7 +72,6 @@
 
     /** {@inheritDoc} */
     @Override protected void afterTestsStopped() throws Exception {
-        stopAllGrids();
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxConcurrentRemoveObjectsTest.java]:
```suggestion

```

Comment (NSAmelchev) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteTxConcurrentRemoveObjectsTest.java]: Also,   **super.afterTestsStopped();** can be deleted here.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/h2/twostep/InOperationExtractPartitionSelfTest.java [hunk 7]
```diff
@@ -116,7 +116,6 @@
     @Override protected void afterTestsStopped() throws Exception {
         orgCache = null;
```
Comment (NSAmelchev) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/h2/twostep/InOperationExtractPartitionSelfTest.java]: Linebreak can be deleted.

**Pull request closed without merging**